app/controllers/UserController.scala

```diff
@@ -0,0 +1,83 @@
+package controllers
+
+import java.security.MessageDigest
+
+import com.google.inject.Inject
+import com.typesafe.scalalogging.StrictLogging
+import dal.Repo
+import models.User
+import play.api.data.{FormError, Form}
+import play.api.data.Forms._
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc._
+import controllers.UserController._
+import sun.misc.BASE64Encoder
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Random
+
+class UserController @Inject()(repo: Repo, val messagesApi: MessagesApi)(implicit ec: ExecutionContext) extends Controller with I18nSupport with StrictLogging {
```

Comment:
Just in case, we need to let the user to use test without registration. if he logins, then he will receive wider access.